### ISSUE TITLE
O3DS: levels don't load properly

### ISSUE THREAD

**Issue by lividhen** [SaturnSH2x2/Sonic-CD-11-3DS#8]
Loading or creating a new save just puts it in a state where there are sprites loaded, the  everything just kind of sits there. 3d works though! (has a couple garbage tiles to the left 😂) 
https://drive.google.com/file/d/1DsnJ5WldVjl_vLsO3KYCn2Ko_gTlaFep/view?usp=sharing

**Comment by SaturnSH2x2** [comment]
- What version of the mobile rsdk file are you using?  Is this the SEGA Forever rsdk, or an earlier version?
- Does this issue still occur with the [Scripts](https://github.com/Rubberduckycooly/Sonic-CD-2011-Script-Decompilation) folder extracted to /3ds/SonicCD?

**Comment by lividhen** [comment]
I'm not quite sure which rsdk I have. It's MD5 is ff439cc19d397b74595ea5a2659b74f0 if you have something to compare it to 😅.
It does still occur with the scripts folder extracted to the SonicCD directory.

Building from the 3ds-hw branch works correctly, but has no audio and has some frame smearing issues on the main menu.

**Comment by SaturnSH2x2** [comment]
The `3ds-hw` branch is deprecated, and is missing several features that are included on the newer branch (i.e. everything that branch introduced got merged into master).  I should've deleted it a while back, but hey.

Out of curiosity, does this issue also occur on the base decomp (i.e. on PC)?  Wondering if this is a 3DS-specific issue or just something weird with the decomp in general.

**Comment by lividhen** [comment]
It works on the windows version of the decompilation.
![image](https://user-images.githubusercontent.com/42221797/129795007-be60c54e-3fea-4baf-96bc-d8beb898d385.png)

**Comment by SaturnSH2x2** [comment]
That's odd.  I'm not entirely sure what's going on, but I'm guessing its a path issue causing certain files not to load.  The 3DS build has had issues with paths not loading properly in the past, either due to case sensitivity or oddly formatted paths, like with #7.

In `settings.ini`, could you set `EngineDebugMode=true` and either send the `log.txt` at `/3ds/SonicCD` and/or screenshot the bottom screen upon loading into Palmtree Panic?

**Comment by Nacheras** [comment]
I have the same error as you with an O3DS, using the rsdk file from the android version and the HW.cia
[log.txt](https://github.com/SaturnSH2x2/Sonic-CD-11-3DS/files/7012999/log.txt)
![1629362545425](https://user-images.githubusercontent.com/50806386/130038084-945f8e42-8856-403a-a57c-8d4f1d7f443f.jpg)

**Comment by lividhen** [comment]
How do you get that log to show?

**Comment by Nacheras** [comment]
@SaturnSH2x2 explained it just before my first post. See it

**Comment by lividhen** [comment]
> 
> 
> @SaturnSH2x2 explained it just before my first post. See it

Oh 🤦
Thank you. missed that.

**Comment by SaturnSH2x2** [comment]
Apparently, scripts aren't being loaded properly on the mobile RSDK:
```
Couldn't load file 'Bytecode/GlobalCode.bin'
Couldn't load file 'Data/Scripts/Players/PlayerObject.txt'
Couldn't load file 'Data/Scripts/Global/StageSetup.txt'
Couldn't load file 'Data/Scripts/Global/HUD.txt'
Couldn't load file 'Data/Scripts/Global/ActFinish.txt'
Couldn't load file 'Data/Scripts/Global/DeathEvent.txt'
Couldn't load file 'Data/Scripts/Players/TailsObject.txt'
Couldn't load file 'Data/Scripts/Global/PauseMenu.txt'
Couldn't load file 'Data/Scripts/Global/Ring.txt'
Couldn't load file 'Data/Scripts/Global/LoseRing.txt'
Couldn't load file 'Data/Scripts/Global/RingSparkle.txt'
Couldn't load file 'Data/Scripts/Global/Monitor.txt'
```

The engine is trying to load `.txt` files instead of the mobile bytecode script files.  Not entirely sure why, but I'll work on fixing this.

**Comment by SaturnSH2x2** [comment]
This problem should be fixed as of dd46a7f93506a2bdfb896c3beb46343f02db660a.  Tested the commit with a mobile RSDK and everything seemed to load.  I apologize for letting a bug this major slip, as I do most of my testing using the Steam RSDK.  I'll push a new release soon with the fix.
